src/lib/support/TypeSerializer.h

```diff
@@ -0,0 +1,102 @@
+/*
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/support/Span.h>
+
+#include <array>
+#include <type_traits>
+
+namespace chip {
+
+template <typename Type, class Enable = void>
+class TypeSerializer;
+
+// specialization for arithmetic types
+template <typename T>
+class TypeSerializer<T, typename std::enable_if<std::is_arithmetic<T>::value>::type>
+{
+public:
+    static constexpr const size_t Space = sizeof(T);
+
+    static T LoadValue(FixedSpan<const char, Space> bytes)
+    {
+        // The span can pointing to arbitrary memory, probably not aligned, so casting data() to pointer may cause bus
+        // error on platform which requires aligned memory read/write. Use memcpy instead
+        T result;
+        ::memcpy(&result, bytes.data(), Space);
```

Comment:
OK, so this serializes/deserializes in native-endian.  That should be _very_ clearly documented.  What are the envisioned use cases here?